packages/babel-traverse/src/path/ancestry.js

```diff
@@ -200,59 +200,3 @@ export function inType() {
 
   return false;
 }
-
-/**
- * Checks whether the binding for 'key' is a local binding in its current function context.
- *
- * Checks if the current path either is, or has a direct parent function that is, inside
- * of a function that is marked for shadowing of a binding matching 'key'. Also returns
- * the parent path if the parent path is an arrow, since arrow functions pass through
- * binding values to their parent, meaning they have no local bindings.
- *
- * Shadowing means that when the given binding is transformed, it will read the binding
- * value from the container containing the shadow function, rather than from inside the
- * shadow function.
- *
- * Function shadowing is acheieved by adding a "shadow" property on "FunctionExpression"
- * and "FunctionDeclaration" node types.
- *
- * Node's "shadow" props have the following behavior:
- *
- * - Boolean true will cause the function to shadow both "this" and "arguments".
- * - {this: false} Shadows "arguments" but not "this".
- * - {arguments: false} Shadows "this" but not "arguments".
- *
- * Separately, individual identifiers can be flagged with two flags:
- *
- * - _forceShadow - If truthy, this specific identifier will be bound in the closest
- *    Function that is not flagged "shadow", or the Program.
- * - _shadowedFunctionLiteral - When set to a NodePath, this specific identifier will be bound
- *    to this NodePath/Node or the Program. If this path is not found relative to the
- *    starting location path, the closest function will be used.
- *
- * Please Note, these flags are for private internal use only and should be avoided.
- * Only "shadow" is a public property that other transforms may manipulate.
- */
-
```

Comment:
Happy to chat if you have suggestions. This PR mostly removes the whole concept of "shadowing" that we had before, in favor of fixing everything up right away, so it's not obvious where we'd put this, or if it would be useful to people at this point.